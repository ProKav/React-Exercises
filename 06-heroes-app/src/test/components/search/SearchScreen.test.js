import React from 'react'
import {SearchScreen} from "../../../components/search/SearchScreen";
import {mount} from "enzyme"
import {MemoryRouter, Route} from "react-router-dom";

describe('Pruebas en <SearchScreen />', () => {

    const wrapper = mount(
        <MemoryRouter initialEntries={['/search']}>
            <Route path="/search" component={SearchScreen} />
        </MemoryRouter>
    )

    test('Debe de mostrarse correctamente con los valores por defecto', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('.alert-info').text().trim()).toBe('Search a Hero')
    })

    test('Debe de mostrar a Batman y el input con el valor de query string', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={SearchScreen} />
            </MemoryRouter>
        )

        expect(wrapper.find('input').prop('value')).toBe('batman')
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar el error si no se encuentra el Hero', () => {

        const hero = 'batman111'

        const wrapper = mount(
            <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
                <Route path="/search" component={SearchScreen} />
            </MemoryRouter>
        )

        expect(wrapper.find('.alert-danger').text().trim()).toBe(`There is not a hero with ${hero} name`)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe de llamar el push del history', () => {

        const hero = 'batman'
        const history = {
            push: jest.fn()
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={[`/search?q=${hero}`]}>
                <Route path="/search" component={() => <SearchScreen history={history} />} />
            </MemoryRouter>
        )
        
        wrapper.find('input').simulate('change'), {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        }
        
        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        })
        
        expect(history.push).toHaveBeenCalledWith('?q=batman')
    })
})