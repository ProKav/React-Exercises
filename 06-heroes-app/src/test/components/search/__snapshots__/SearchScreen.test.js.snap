// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchScreen /> Debe de mostrar a Batman y el input con el valor de query string 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find Your Hero"
          type="text"
          value="batman"
        />
        <button
          className="btn btn-block btn-outline-primary m-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="card mb-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row g-0"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src=""
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                Mas...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchScreen /> Debe de mostrar el error si no se encuentra el Hero 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find Your Hero"
          type="text"
          value="batman111"
        />
        <button
          className="btn btn-block btn-outline-primary m-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is not a hero with 
        batman111
         name
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en <SearchScreen /> Debe de mostrarse correctamente con los valores por defecto 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find Your Hero"
          type="text"
          value=""
        />
        <button
          className="btn btn-block btn-outline-primary m-1"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a Hero
      </div>
    </div>
  </div>
</div>
`;
